packages/p2p-v2/src/pages/my-profile/screens/MyProfileAdDetails/MyProfileAdDetails.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useAdvertiserInfo, useAdvertiserUpdate } from '@deriv/api';
+import { Button, FullPageMobileWrapper, TextArea } from '../../../../components';
+import { useDevice } from '../../../../hooks';
+import './MyProfileAdDetails.scss';
+
+const MyProfileAdDetails = () => {
+    const { data: advertiserInfo, isLoading } = useAdvertiserInfo();
+    const { mutate: updateAdvertiser } = useAdvertiserUpdate();
+    const [contactInfo, setContactInfo] = useState('');
+    const [advertDescription, setAdvertDescription] = useState('');
+    const { isMobile } = useDevice();
+
+    const hasUpdated = useMemo(() => {
+        return (
+            contactInfo !== advertiserInfo?.contact_info ||
+            advertDescription !== advertiserInfo?.default_advert_description
+        );
+    }, [advertiserInfo?.contact_info, advertiserInfo?.default_advert_description, contactInfo, advertDescription]);
+
+    useEffect(() => {
+        setContactInfo(advertiserInfo?.contact_info || '');
+        setAdvertDescription(advertiserInfo?.default_advert_description || '');
+    }, [advertiserInfo]);
+
+    const submitAdDetails = () => {
+        updateAdvertiser({
+            contact_info: contactInfo,
+            default_advert_description: advertDescription,
+        });
+    };
+
+    if (isLoading) return <h1>Loading...</h1>;
+
+    if (isMobile) {
+        return (
+            <FullPageMobileWrapper
+                renderFooter={() => (
+                    <Button disabled={!hasUpdated} isFullWidth onClick={submitAdDetails}>
+                        Save
+                    </Button>
+                )}
+                renderHeader={() => <h1 className='p2p-v2-my-profile-ad-details__header'>Ad Details</h1>}
+            >
+                <div className='p2p-v2-my-profile-ad-details'>
+                    <TextArea
+                        onChange={e => setContactInfo(e.target.value)}
+                        placeholder='My contact details'
+                        value={advertiserInfo?.contact_info || ''}
+                    />
+                    <TextArea
+                        label='This information will be visible to everyone.'
+                        onChange={e => setAdvertDescription(e.target.value)}
+                        placeholder='Instructions'
+                        value={advertiserInfo?.default_advert_description || ''}
+                    />
```

Comment:
since this is the same as below, can we extract this out 👀 of course leave it in the same file 👍

Reply:
It'll lead to more lines of code though, extracting it into a separate component means duplicating the state updates and hook calls 😢 
It's just a few components difference too though

Reply:
I dont think itll create more code tbh, im thinking something like this:

```suggestion
const TextAreaDetails = ({onChangeContactInfo, contactInfoValue, onChangeAdvertDescription, 
        advertDescriptionValue }) => (
            <TextArea
                onChange={onChangeContactInfo}
                placeholder='My contact details'
                value={contactInfoValue}
            />
            <TextArea
                label='This information will be visible to everyone.'
                onChange={onChangeAdvertDescription}
                placeholder='Instructions'
                value={advertDescriptionValue}
            />
)
```

Then in the div:

```suggestion
            <TextAreaDetails
                    onChangeContactInfo={e => setContactInfo(e.target.value)}
                    contactInfoValue={advertiserInfo?.contact_info || ''}
                    onChangeAdvertDescription={e => setAdvertDescription(e.target.value)}
                    advertDescriptionValue={advertiserInfo?.default_advert_description || ''}
             />
```